src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
 
+export interface Figure {
+  shape: Shape,
+  color: 'red' | 'green' | 'blue',
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape = Shape.triangle;
 
-}
+  constructor(
+    public color: Figure['color'],
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Not valid value');
+    }
+
+    if (this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.a + this.c <= this.b) {
+      throw new Error('sides 1, 2 and 3 can not form a triangle');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const result = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
 
+    return Math.floor(100 * result) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = Shape.circle;
+
+  constructor(
+    public color: Figure['color'],
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('Not valid value');
+    }
+  }
 
+  getArea(): number {
+    const area = Math.PI * (this.radius ** 2);
+    const result = Math.floor(area * 100) / 100;
+
+    return result;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape = Shape.rectangle;
```

Comment:
you can omit writing public, it will be public by default

```suggestion
  shape = Shape.rectangle;
```